### PR TITLE
[DOCFIX] Architecture and Data Flow 

### PR DESCRIPTION
**Pull request by apc999**
it was named Key Concepts

### CODE REVIEW

File: docs/en/Architecture-DataFlow.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+If the data is not available within the Alluxio space, a cache miss occurs and the application will have to read the data from the under storage. The Alluxio client will delegate the read to a local worker and the worker will read the data from the under storage. Up to the client, the worker may cache the data locally for future reads. Cache misses generally cause the largest delay because the application has to wait until the data is fetched from the under storage. A cache miss typically happens when the data is read the first time.
+
+when Alluxio client reads a block that is not locally available, it will read and cache the block entirely on the local worker even if only part of the block is requested by the client. If there is no such a local worker, no cache will happen. For certain workloads and file formats (e.g., running SQL queries on ORC and Parquet files), the data reads are often not sequential. In this case, having partial caching will help users cache the input data into Alluxio even if only the footer or header of these files are read, significantly improving the performance of the following queries on the same input data.
```
Comment (aaudiber) [inline docs/en/Architecture-DataFlow.md]: do remote reads not do partial caching?

File: docs/en/Architecture-DataFlow.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+In Alluxio versions 1.7 and above, partial caching is always enabled and handled on the server side in the background. The caching process is completely transparent to the client. The client will read data as normal and signal to the worker which blocks it reads asynchronously. The worker will fetch these blocks from the under storage or other Alluxio workers asynchronously. Duplicate requests caused by concurrent readers will be consolidated on the worker and result in caching the block once. Partial caching is not on the critical path, but may still impact performance if the network bandwidth between Alluxio and the under storage system is a bottleneck.
```
Comment (aaudiber) [inline docs/en/Architecture-DataFlow.md]: Our docs are versioned and 1.7 came out almost a year ago. Should we simplify this section to only talk about the 1.7/1.8 behavior?

File: docs/en/Architecture-DataFlow.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+![Cache Miss data flow]({{ site.baseurl }}/img/dataflow-cache-miss.gif)
+
+#### Read Misses Alluxio but do not Cache
+
+It’s possible to turn off caching in Alluxio and have the client read directly from the under storage by setting the property `alluxio.user.file.readtype.default` in the client to `NO_CACHE`.
```
Comment (aaudiber) [inline docs/en/Architecture-DataFlow.md]: the client still reads through the worker, right?

File: docs/en/Architecture-DataFlow.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+### Overview
+
+Alluxio is not a persistent storage system. Instead,
+Alluxio serves as a new data access layer in the ecosystem, residing between any persistent storage
+systems (such as Amazon S3, Microsoft Azure Object Store, Apache HDFS or OpenStack Swift)
+and computation frameworks
+(such as Apache Spark, Presto or Hadoop MapReduce).
+There are multiple benefits to have this layer in the stack:
+
+- For user applications and computation frameworks, Alluxio can provide them fast storage, facilitating data sharing and locality between jobs, regardless of whether they are running on the same computation engine. As a result, Alluxio can serve the data at memory speed when data is local, or the computation cluster network speed when data is in Alluxio. Data is only read once from the under storage system on the first time it’s accessed. Therefore, the data access can be significantly accelerated when the access to the under storage is not fast. To achieve the best performance, Alluxio is recommended
```
Comment (aaudiber) [inline docs/en/Architecture-DataFlow.md]: can we split these long lines into shorter lines? We follow that style in most of our markdown docs, and it makes it easier to read diffs and write review comments

Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: done

File: docs/en/Architecture-DataFlow.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+#### Standby Masters
+Standby masters read journals written by the primary master to keep their own copies of master state up-to-date. They also write journal checkpoints for faster recovery in the future. They do not process any requests from other Alluxio components.
+
+### Worker
+
+![Alluxio worker]({{ site.baseurl }}/img/architecture-worker.png)
+
+Alluxio workers are responsible for managing user-configurable local resources allocated to Alluxio (e.g. memory, SSDs, HDDs etc.). Alluxio workers store data as blocks and serve client requests that read or write data by reading or creating new blocks within their local resources. Workers are only responsible for managing blocks; the actual mapping from files to blocks is only stored by the master.
+
+Also, Alluxio workers perform data operations on the under store (e.g. data transfer or under store metadata operations). This brings two important benefits: The data read from the under store can be stored in the worker and be available immediately to other clients, and the client can be lightweight and does not depend on the under storage connectors.
```
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: done

File: docs/en/Architecture-DataFlow.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+Because RAM usually offers limited capacity, blocks in a worker can be evicted when space is full. Workers employ eviction policies to decide which data to keep in the Alluxio space. For more on this topic, please check out the documentation for [Tiered Storage]({{ site.baseurl }}{% link en/advanced/Alluxio-Storage-Management.md %}#multiple-tier-storage).
+
+
+### Client
+
+The Alluxio client provides users a gateway to interact with the Alluxio servers. It initiates communication with the primary master to carry out metadata operations and with workers to read and write data that is stored in Alluxio. It also provides a native filesystem API in Java, and supports multiple client languages including REST, Go and Python. In addition to that, Alluxio also supports APIs that are compatible with HDFS API as well as Amazon S3 API.
```
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: rephrased.

File: docs/en/Architecture-DataFlow.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+## Data flow
+
+This section describes the behavior of common read and write scenarios based on a typical Alluxio configuration as described above: Alluxio is colocated with the compute framework and and applications and the persistent storage system is either a remote storage cluster or cloud-based storage.
+
+### Read
+
+Sitting between the under storage and computation framework, Alluxio can serve as a caching layer for data reads. This subsection introduces different caching scenarios and their implications on performance.
+
+#### Read Hits Alluxio Cache on a Local Worker
```
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: I done

File: docs/en/Architecture-DataFlow.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+this topic, please check out the instructions on [running Alluxio on Docker]({{ site.baseurl }}{% link en/deploy/Running-Alluxio-On-Docker.md %}).
+
+Also note that Alluxio can manage other storage media (e.g. SSD, HDD) in addition to memory, so local data access speed may vary depending on the local storage media. To learn more about this topic, please refer to the [tiered storage document]({{ site.baseurl }}{% link en/advanced/Alluxio-Storage-Management.md %}#multiple-tier-storage).
+
+![Data Flow of Read from a Local Worker]({{ site.baseurl }}/img/dataflow-local-cache-hit.gif)
+
+#### Read Hits Alluxio Cache on a Remote Worker
+
+When the data is not available in a local Alluxio worker but resides in another Alluxio worker for the cluster, the Alluxio client will read from the remote worker. For example, if the Alluxio client checks with the Alluxio master and finds that the data is available on a remote Alluxio worker, the local worker will read the data from the remote worker and forward the data to the client.
```
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: rephrased to reflect the latest procedure of async cache

File: docs/en/Architecture-DataFlow.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+In addition to returning the data to the client, the worker will also write a copy locally so that future reads of the same data can be served locally from memory. Remote cache hits provide network-speed data reads. Alluxio prioritizes reading from remote workers over reading from under storage because the network speed between Alluxio workers is typically faster than the speed between Alluxio workers and the under storage.
+
+![Data Flow of Read from a Remote Worker]({{ site.baseurl }}/img/dataflow-remote-cache-hit.gif)
+
+
+#### Read Misses Alluxio Cache and Triggers Cache
+
+If the data is not available within the Alluxio space, a cache miss occurs and the application will have to read the data from the under storage. The Alluxio client will delegate the read to a local worker and the worker will read the data from the under storage. Up to the client, the worker may cache the data locally for future reads. Cache misses generally cause the largest delay because the application has to wait until the data is fetched from the under storage. A cache miss typically happens when the data is read the first time.
```
[the remark this answers is not in the document]
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: fixed

File: docs/en/Architecture-DataFlow.md (hunk 1 above)
Comment (apc999) [reply docs/en/Architecture-DataFlow.md]: re-wrote the async caching part.

**Comment by apc999** [comment]
@aaudiber ptal

**Pull request merged**